staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -456,6 +478,11 @@ func writePreferences(configAccess ConfigAccess, newPrefs clientcmdapi.Preferenc
 
 		if !reflect.DeepEqual(currConfig.Preferences, newPrefs) {
 			currConfig.Preferences = newPrefs
+
+			if lockErrors[file] != nil {
```

Comment:
move this above the getConfigFromFile call

Reply:
and also add before the getConfigFromFile in the IsExplicitFile() block above

Reply:
Done and done: ecf3b83e88ef4c4791889f922514cf459d3b3058